pyquil/analytics.py

```diff
@@ -0,0 +1,310 @@
+                gate_counts['Z'] + \
+                gate_counts['S'] + \
+                gate_counts['T'] + \
+                gate_counts['PHASE']
+    results['Two Qubit Gates'] = \
+                gate_counts['RX'] + \
+                gate_counts['RY'] + \
+                gate_counts['RZ'] + \
+                gate_counts['CNOT'] + \
+                gate_counts['CPHASE00'] + \
+                gate_counts['CPHASE01'] + \
+                gate_counts['CPHASE10'] + \
+                gate_counts['CPHASE'] + \
+                gate_counts['SWAP'] + \
+                gate_counts['ISWAP'] + \
+                gate_counts['PSWAP']
+    results['Control Instructions'] = \
+                gate_counts['RESET'] + \
+                gate_counts['NOP'] + \
+                gate_counts['WAIT'] + \
+                gate_counts['HALT']
+    results['Classical Instructions'] = \
+                gate_counts['TRUE'] + \
+                gate_counts['FALSE'] + \
+                gate_counts['NOT'] + \
+                gate_counts['AND'] + \
+                gate_counts['OR'] + \
+                gate_counts['MOVE'] + \
+                gate_counts['EXCHANGE'] + \
+                gate_counts['JUMP-WHEN'] + \
+                gate_counts['JUMP-UNLESS']
+    results['Three+ Qubit Gates'] = \
+                gate_counts['CCNOT'] + \
+                gate_counts['CSWAP']
+    results['Elements in DEFGATES'] = \
+                gate_counts['DEFGATE']
+    return results
+
+def analyze(program_creator, input_list, should_range=False, to_include=None, remove_zeros=False):
+    '''
+    Analyzes a program by compiling properties of the pyQuil program for different inputs. Provides
+    a wide range of configurations to assist in analyzing specific programs
+    :param program_creator: A method to accept program parameters and create the Quil program (i.e
+                            a method such as qft from grove.fourier).
+    :param input_list: A list of inputs to be inputted into the program creator. These may be tuples
+                       of parameters, or something as simple as a list [0, 1, 2 ...] indicating the
+                       number of qubits to use in a program.
+    :param should_range: (Optional) If true, takes each value in n_list, assumes that it is an integer,
+                         and replaces it with range(that integer). This is useful in situations such as
+                         using 1, 2, 3, ... qubits without indicating the range of indices for these
+                         qubits. Default is False.
+    :param to_include: (Optional) A list of properties to analyze with your program creator. Default is
+                       to include every parameter.
+    :param remove_zeros: (Optional) If set to True, any properties where the value is 0 for all inputs
+                         of the program are left out of the final result. Default is False.
+    :return: A tuple containing multiple analysis results:
```

Comment:
again here I think it would be best to sit down and think about the data structure you want to construct, rather than defaulting to a funky set of tuples